api/src/accounts.rs

```diff
@@ -251,6 +296,83 @@ impl Account {
         }
     }
 
+    pub fn balance(&self, asset_type: AssetType, accept_type: &AcceptType) -> BasicResultWith404<u64> {
+        let (fa_metadata_address, mut balance) = match asset_type {
+            AssetType::Coin(move_struct_tag) => {
+                let coin_store_type_tag = StructTag::from_str(&format!("0x1::coin::CoinStore<{}>", move_struct_tag.to_string())).map_err(|err| {
```

Comment:
Do you want to validate the move struct type first?

Ensure that we're not getting balances of say `u8`?

Also, ensure it's limited in parsing